File: Python/Auto_Exposure.py
import cv2

# 讀取影像
image = cv2.imread("image.jpg")

# 創建一個窗口，用來顯示影像
cv2.namedWindow("image")
cv2.imshow("image", image)

# 設定自動曝光演算法的參數
# 在這個例子中，我們使用了曝光模式為自動曝光，並且指定了自動曝光限制為0.25秒
# 具體的，自動曝光演算法會根據影像中的亮度和細節信息來自動調整曝光時間
# 但是，由於我們指定了自動曝光限制為0.25秒，因此如果曝光時間大於0.25秒，曝光演算法就會被限制在0.25秒
cap = cv2.VideoCapture(0)
cap.set(cv2.CAP_PROP_EXPOSURE, cv2.CAP_PROP_AUTO_EXPOSURE)
cap.set(cv2.CAP_PROP_EXPOSURE, 0.25)

# 讀取影像
ret, image = cap.read()

# 創建一個窗口，用來顯示影像
cv2.namedWindow("image")
cv2.imshow("image", image)

# 等待按鍵
cv2.waitKey()

File: C++/Auto_Exposure.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
  // Open the default camera
  VideoCapture cap(0);

  if (!cap.isOpened()) {
    cerr << "ERROR: Could not open camera" << endl;
    return 1;
  }

  // Set the camera to auto exposure
  cap.set(CAP_PROP_AUTO_EXPOSURE, 1.0);

  while (true) {
    Mat frame;

    // Read a frame from the camera
    cap >> frame;

    if (frame.empty()) {
      cerr << "ERROR: Could not grab frame from camera" << endl;
      break;
    }

    // Show the frame
    imshow("Camera", frame);

    // Check for user input
    if (waitKey(1) >= 0) {
      break;
    }
  }

  return 0;
}
// This code will open the default camera and set it to auto exposure mode. It will then continuously grab frames from the camera and display them on the screen. When the user presses any key, the program will exit.

// You will need to have the OpenCV library installed on your computer to compile and run this code. You can learn more about how to install OpenCV on the OpenCV website: https://opencv.org/




